src/main/java/com/iota/iri/Milestone.java

```diff
@@ -55,33 +57,31 @@
 
     private final Set<Hash> analyzedMilestoneCandidates = new HashSet<>();
 
-    public Milestone(final Tangle tangle,
-                     final Hash coordinator,
-                     final Snapshot initialSnapshot,
-                     final TransactionValidator transactionValidator,
-                     final boolean testnet,
-                     final MessageQ messageQ,
-                     final int numOfKeysInMilestone,
-                     final int milestoneStartIndex,
-                     final boolean acceptAnyTestnetCoo
+    public Milestone(Tangle tangle,
+                     TransactionValidator transactionValidator,
+                     MessageQ messageQ,
+                     ConsensusConfig config,
+                     Snapshot initialSnapshot
                      ) {
         this.tangle = tangle;
-        this.coordinator = coordinator;
-        this.latestSnapshot = initialSnapshot;
         this.transactionValidator = transactionValidator;
-        this.testnet = testnet;
         this.messageQ = messageQ;
-        this.numOfKeysInMilestone = numOfKeysInMilestone;
-        this.milestoneStartIndex = milestoneStartIndex;
+        this.latestSnapshot = initialSnapshot;
+
+        //configure
+        this.testnet = config.isTestnet();
+        this.coordinator = new Hash(config.getCoordinator());
+        this.numOfKeysInMilestone = config.getNumberOfKeysInMilestone();
+        this.milestoneStartIndex = config.getMilestoneStartIndex();
         this.latestMilestoneIndex = milestoneStartIndex;
         this.latestSolidSubtangleMilestoneIndex = milestoneStartIndex;
-        this.acceptAnyTestnetCoo = acceptAnyTestnetCoo;
+        this.acceptAnyTestnetCoo = !config.isValidateTestnetMilestoneSig();
```

Comment:
how about refactoring this & `testnet` into a `testnetMode` flag with values `{disabled/mainnet, enforcing, permissive}` or similar

Reply:
Perhaps it is a good idea. Not for this PR for sure.
Tagging @geminoz @jakubcech  to see if they think the idea of giving different permissions on a testnet with one flag is appealing.